### PR TITLE
Fieldnames to propertynames

### LINKED ISSUE #191: RealFourierArray autocompletion isn't bypassed properly
It's supposed to only show the non-underscored properties (as per the fieldnames overload), but it doesn't.

### PR DESCRIPTION
**Pull request by mfherbst**
Fix #191

### CODE REVIEW

File: src/RealFourierArray.jl [hunk 1]
```diff
@@ -49,8 +49,11 @@ end
 check_real(A::RealFourierArray{Treal, T}) where {Treal, T <: Real} = nothing
 check_real(A::RealFourierArray) = check_real(A.real)
 
-# lie blatantly to the user, as recommended by the docs
-Base.fieldnames(RealFourierArray) = (:basis, :real, :fourier)
+function Base.propertynames(array::RealFourierArray, private=false)
+    ret = [:basis, :real, :fourier]
+    private && append!(ret, fieldnames(RealFourierArray))
```
Comment (antoine-levitt) [inline src/RealFourierArray.jl:54]: boo mutation :-p
`private ? append(ret, fieldnames(RealFourierArray)) : ret` ?

**Comment by antoine-levitt** [comment]
Huh, apparently I misread the docs. I just posted it here so I wouldn't forget but you're too damn quick :-p thanks!

File: src/RealFourierArray.jl (hunk 1 above)
Comment (mfherbst) [reply src/RealFourierArray.jl:54]: There is no `append` ... so if you don't want to repeat ret I don't see a way without mutation.

Comment (antoine-levitt) [reply src/RealFourierArray.jl]: Really, that's weird. `[ret..., fieldnames(RealFourierArray)...]`

Comment (mfherbst) [reply src/RealFourierArray.jl]: Really, what's the point? This function is not performance-critical and what you propose either allocates twice as well (if ret is an array) or is not type-stable (once returns array, once tuple if ret is a tuple). Also it's harder to read. I'm not convinced.

Comment (antoine-levitt) [reply src/RealFourierArray.jl]: Ret should not be a tuple? But yeah I'm fine with what you wrote, it's just my part-time crusade against mutation.

**Comment by antoine-levitt** [comment]
@gkemlin

File: src/RealFourierArray.jl (hunk 1 above)
Comment (mfherbst) [reply src/RealFourierArray.jl]: No. either tuple or vector. Both is allowed explicitly.

**Pull request merged**